proposals/0000-local-do.rst

```diff
@@ -236,38 +261,61 @@ Related work
   * A more recent idea is `supermonads <http://www.cs.nott.ac.uk/~psznhn/Publications/jfp2018.pdf>`_, which solves the type inference issue using a plugin
 
-  It requires somewhat less work (“only” a plugin, rather than a change in GHC's compiler, at least it's more modular), and is more automatic, as the correct functions are picked automatically from the type. But there is no way that this will capture all the desired notion: some restrictions need be imposed for the sake of type inference. Note as well that this proposal doesn't preclude an automatic approach when appropriate: simply import your very automatic builder in scope, and all the do-expressions without an explicit builder will use this.
-
-* There is a way to emulate ``-XLocalDo`` in current GHC using ``-XRecordWildcards``: have no ``(>>=)`` and such in scope, and import a builder with ``Builder {..} = builder``. It is used in `linear-base <https://github.com/tweag/linear-base/blob/0d6165fbd8ad84dd1574a36071f00a6137351637/src/System/IO/Resource.hs#L119-L120>`_. This is not a very good solution: it is rather a impenetrable idiom, and, if a single function uses several builders, it yields syntactic contortion (which is why shadowing warnings are deactivated `here <https://github.com/tweag/linear-base/blob/0d6165fbd8ad84dd1574a36071f00a6137351637/src/System/IO/Resource.hs#L1>`_)
+  It requires somewhat less work (“only” a plugin, rather than a change in GHC's compiler, at least it's more modular), and is more automatic, as the correct functions are picked automatically from the type. But there is no way that this will capture all the desired notion: some restrictions need be imposed for the sake of type inference.
 
-* Instead of changing the ``Prelude``, the standard builder could be
-  hosted in a separate module (such as ``Ghc.LocalDo``), and the
-  programmer could ``import Ghc.LocalDo`` when they use ``-XLocalDo``.
-
-* An alternative to the ``@<aexpr>`` notation would be to use implicit parameters, somehow. But it's unclear how exactly it would look.
+* There is a way to emulate ``-XQualifiedDo`` in current GHC using ``-XRecordWildcards``: have no ``(>>=)`` and such in scope, and import a builder with ``Builder {..} = builder``. It is used in `linear-base <https://github.com/tweag/linear-base/blob/0d6165fbd8ad84dd1574a36071f00a6137351637/src/System/IO/Resource.hs#L119-L120>`_. This is not a very good solution: it is rather a impenetrable idiom, and, if a single function uses several builders, it yields syntactic contortion (which is why shadowing warnings are deactivated `here <https://github.com/tweag/linear-base/blob/0d6165fbd8ad84dd1574a36071f00a6137351637/src/System/IO/Resource.hs#L1>`_)
 
 Extensions
 ~~~~~~~~~~
 
-Comprehension
-+++++++++++++
+Qualified do with parameters
+++++++++++++++++++++++++++++
+
+At some point, this proposal could be complemented with another extension
```

Comment:
Please let's not suggest adding extra extensions. We should just change the meaning of this one, instead.

Reply:
Done.